pkg/apis/infra/v1alpha1/cluster_types.go

```diff
@@ -0,0 +1,226 @@
+	// Defaults to 10.0.0.0/16.
+	// +optional
+	CidrBlock *string `json:"cidrBlock"`
+	// Subnets is the list of subnets to be used when the provider creates a managed VPC.
+	Subnets []SubnetConfig `json:"subnets"`
+}
+
+type SubnetConfig struct {
+	// ID defines a unique identifier to reference this resource.
+	ID string `json:"id"`
+	// CidrBlock is the CIDR block to be used when the provider creates a managed subnet.
+	CidrBlock *string `json:"cidrBlock"`
+}
+
+type CNIConfig struct {
+	// Cilium is the configuration for the Cilium CNI.
+	// +optional
+	Cilium *CiliumConfig `json:"cilium,omitempty"`
+}
+
+type CiliumConfig struct {
+}
+
+type MasterConfig struct {
+	NodeConfig `json:",inline"`
+}
+
+type WorkerConfig struct {
+	NodeConfig `json:",inline"`
+	Strategy   *WorkerNodeStrategy `json:"strategy,omitempty"`
+}
+
+type WorkerNodeStrategy struct {
+	Type          WorkerNodeStrategyType           `json:"type,omitempty"`
+	RollingUpdate *RollingUpdateWorkerNodeStrategy `json:"rollingUpdate,omitempty"`
+}
+
+type WorkerNodeStrategyType string
+
+const (
+	// RollingUpdateWorkerNodeStrategyType replaces old machines by new one using rolling update.
+	RollingUpdateWorkerNodeStrategyType WorkerNodeStrategyType = "RollingUpdate"
+
+	// OnDeleteWorkerNodeStrategyType replaces old machines when the deletion of the asssoicated machines are completed.
+	OnDeleteWorkerNodeStrategyType WorkerNodeStrategyType = "OnDelete"
+)
+
+type RollingUpdateWorkerNodeStrategy struct {
+	// MaxUnavailable is the maximum number of nodes that can be unavailable during the update.
+	MaxUnavailable *intstr.IntOrString `json:"maxUnavailable,omitempty"`
+	// MaxSurge is the maximum number of nodes that can be created above the desired number of nodes during the update.
+	MaxSurge *intstr.IntOrString `json:"maxSurge,omitempty"`
+	// DeletePolicy defines the policy used to identify nodes to delete when downscaling.
+	// Valid values are "Random", "Newest" and "Oldest".
+	// Defaults to "Newest".
+	// +kubebuilder:validation:Enum=Random;Newest;Oldest
+	// +optional
+	DeletePolicy *string `json:"deletePolicy,omitempty"`
+}
+
+// NodeConfig defines the configuration for the node.
+type NodeConfig struct {
+	MachineConfig    `json:",inline"`
+	KubernetesConfig `json:",inline"`
+}
+
+// MachineConfig defines the configuration for the machine.
+type MachineConfig struct {
+	// Replicas is the number of replicas of the machine.
+	Replicas int `json:"replicas"`
+	// AWS is the configuration for the AWS machine.
+	AWS *AWSMachineConfig `json:"aws,omitempty"`
+}
+
+// KubernetesConfig defines the configuration for the Kubernetes node.
+type KubernetesConfig struct {
+	// Labels is the set of labels to apply to the nodes.
+	Labeles map[string]string `json:"labels,omitempty"`
+	// Taints is the set of taints to apply to the nodes.
+	Taints []corev1.Taint `json:"taints,omitempty"`
+}
+
+// AWSMachineConfig defines the configuration for the AWS machine.
+type AWSMachineConfig struct {
```

Comment:
isn't there a field for flavor and image?

Reply:
`InstanceType` represent for flavor

AWS cluster api provider use [ImageLookupFormat](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/f78e519145f184cd612a9854400d249fc3edaf1d/api/v1beta2/awsmachine_types.go#L69), we can add this later.